Source/Core/Core/AchievementManager.cpp

```diff
@@ -129,15 +129,82 @@ void AchievementManager::LoadGameByFilenameAsync(const std::string& iso_path,
 
     const auto fetch_game_data_response = FetchGameData();
     m_is_game_loaded = fetch_game_data_response == ResponseType::SUCCESS;
+
+    // Claim the lock, then queue the fetch unlock data calls, then initialize the unlock map in
+    // ActivateDeactiveAchievements. This allows the calls to process while initializing the
+    // unlock map but then forces them to wait until it's initialized before making modifications to
+    // it.
+    m_lock.lock();
+    LoadUnlockData([](ResponseType r_type) {});
+    ActivateDeactivateAchievements();
+    m_lock.unlock();
+    ActivateDeactivateLeaderboards();
+    ActivateDeactivateRichPresence();
+
     callback(fetch_game_data_response);
   });
 }
 
+void AchievementManager::LoadUnlockData(const ResponseCallback& callback)
+{
+  m_queue.EmplaceItem([this, callback] {
+    const auto hardcore_unlock_response = FetchUnlockData(true);
+    if (hardcore_unlock_response != ResponseType::SUCCESS)
+    {
+      callback(hardcore_unlock_response);
+      return;
+    }
+
+    callback(FetchUnlockData(false));
+  });
+}
+
+void AchievementManager::ActivateDeactivateAchievements()
+{
+  for (unsigned ix = 0; ix < m_game_data.num_achievements; ix++)
+  {
+    if (m_unlock_map.count(m_game_data.achievements[ix].id) == 0)
+      m_unlock_map[m_game_data.achievements[ix].id] = {.game_data_index = ix};
+    ActivateDeactivateAchievement(m_game_data.achievements[ix].id);
+  }
+}
+
+void AchievementManager::ActivateDeactivateLeaderboards()
+{
+  for (unsigned int ix = 0; ix < m_game_data.num_leaderboards; ix++)
+  {
+    // TODO lillyjade: replace false with the Hardcore flag
+    if (m_is_game_loaded && Config::Get(Config::RA_LEADERBOARDS_ENABLED) && false)
+    {
+      if (rc_runtime_get_lboard(&m_runtime, m_game_data.leaderboards[ix].id) == nullptr)
```

Comment:
Maybe extract the leaderboard entry into a reference like:

```cpp
const auto& very_cool_name = m_game_data.leaderboards[ix];
```

so the long indexing doesn't need to be repeated below as well.